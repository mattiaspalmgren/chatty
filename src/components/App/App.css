@import "../../common/border.css";

.App {
    composes: padding-medium wrapper-width from "../../common/layout.css";
    composes: serif from "../../common/typography.css";
    border: var(--dark-border);
}