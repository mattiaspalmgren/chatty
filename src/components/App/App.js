import React from "react";
import gridStyle from "../../common/grid.css";
import style from "./App.css";
import UserForm from "../../containers/UserForm";
import UserList from "../../containers/UserList";
import ChannelList from "../../containers/ChannelList";
import ChannelForm from "../../containers/ChannelForm";

const App = () => (
    <div className={style.App}>
      <h1>Chatty</h1>
      <div className={gridStyle.gridColumns}>
        <div>
            <UserForm/>
            <UserList/>
        </div>
        <div>
            <ChannelForm/>
            <ChannelList/>
        </div>
      </div>
    </div>
);

export default App;