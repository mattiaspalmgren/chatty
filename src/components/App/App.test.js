import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

import state from '../../state.js';

it('render user shallow correctly', () => {
  const tree = shallow(<App state={state} />);
  expect(tree).toMatchSnapshot();
});
