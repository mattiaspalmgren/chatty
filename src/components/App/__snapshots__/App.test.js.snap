// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render user shallow correctly 1`] = `
<div
  className="App"
>
  <Header />
  <div
    className="gridColumns"
  >
    <div>
      <InputForm
        inputForm={
          Object {
            "fields": Array [
              "name",
              "email",
            ],
            "onSubmit": [Function],
            "title": "Add user",
          }
        }
      />
      <User
        key="id"
        user={
          Object {
            "email": "alice@chatter.com",
            "id": "id",
            "name": "Alice",
            "subscriptions": Array [
              "channel1",
              "channel2",
            ],
          }
        }
      />
    </div>
    <div>
      <InputForm
        inputForm={
          Object {
            "fields": Array [
              "name",
            ],
            "onSubmit": [Function],
            "title": "Add channel",
          }
        }
      />
      <Channel
        channel={
          Object {
            "id": "id",
            "name": "Official",
            "subscribers": Array [
              "user1",
              "user2",
            ],
          }
        }
        key="id"
      />
    </div>
  </div>
</div>
`;
